recipes/openvdb/all/conanfile.py

```diff
@@ -27,91 +30,202 @@ class OpenVDBConan(ConanFile):
+        "nanovdb_with_openvdb": [True, False],
+        "nanovdb_with_tbb": [True, False],
     }
     default_options = {
         "shared": False,
         "fPIC": True,
         "with_blosc": True,
         "with_zlib": True,
         "with_log4cplus": False,
-        "with_exr": False,
         "simd": None,
+        "delayed_load": True,
+        "nanovdb": False,        
+        "nanovdb_with_openvdb": True,
+        "nanovdb_with_tbb": True,
     }
 
-    generators = "cmake", "cmake_find_package"
+    @property
+    def _has_sse_avx(self):
+        return self.settings.arch in ["x86", "x86_64"]
+
+    @property
+    def _needs_boost(self):
+        return True
+        # Will be fixed from upstream
+        # return Version(self.version) < "10.0.0" or self.options.delayed_load
 
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _needs_openexr(self):
+        return Version(self.version) < "8.1.0"
 
     @property
-    def _build_subfolder(self):
-        return "build_subfolder"
+    def _minimum_cpp_standard(self):
+        return 14
 
     @property
     def _compilers_min_version(self):
-        return {
-            "msvc": "191",
-            "Visual Studio": "15",  # Should we check toolset?
-            "gcc": "6.3.1",
-            "clang": "3.8",
-            "apple-clang": "3.8",
-            "intel": "17",
-        }
+        if Version(self.version) == "8.0.1":
+            return {
+                "msvc": "191",
+                "Visual Studio": "15",  # Should we check toolset?
+                "gcc": "6.3.1",
+                "clang": "3.8",
+                "apple-clang": "3.8",
+                "intel": "17",
+            }
+        elif Version(self.version) == "10.0.1":
+            return {
+                "msvc": "1928",
+                "Visual Studio": "16.8.1",  # Should we check toolset?
+                "gcc": "9.3.1",
+                "clang": "5",
+                "apple-clang": "5",
+                "intel": "19",
+            }
 
     def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
+        if not self._has_sse_avx:
+            del self.options.simd
+        
+        if Version(self.version) < "9.0.0":
+            del self.options.nanovdb
+            del self.options.nanovdb_with_openvdb
+            del self.options.nanovdb_with_tbb
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def _cmake_new_enough(self, required_version):
```

Comment:
This idiom is no longer necessary, we can remove ``_cmake_new_enough``, there is a feature in Conan 2.0 that implements this, and the new ``cmake`` packages in Conan center have pre-compiled binaries would be much faster, so a simpler:
```python
def build_requirements(self):
     self.tool_requires("cmake/[>=3.18 <4.0]")
```
The simpler the better. We will be recommending this way and moving existing recipes in ConanCenter that already use this approach to this simpler approach.